### ISSUE TITLE
[Question] Why are we using 'createElement' instead of JSX?

### ISSUE THREAD

**Issue by MichaelDeBoey** [inertiajs/inertia-react#17]
Since @sebastiandedeyne made the initial implementation, I think he's the best man to answer this question.

Can you explain why we're using the `createElement` functions instead of JSX directly?

1) In `App.js` we're doing
https://github.com/inertiajs/inertia-react/blob/e921a05a627f8abbd430b8a35aecc9d5027f98b8/src/App.js#L30-L34
which could be 
```js
return (
  <PageContext.Provider value={}>
    {children ? children(pageContents) : pageContents}
  </PageContext.Provider>
)
```

2) In `Link.js` we're doing
https://github.com/inertiajs/inertia-react/blob/e921a05a627f8abbd430b8a35aecc9d5027f98b8/src/Link.js#L24
which could be 
```js
return <a {...props} href={href} onClick={visit}>{children}</a>
```

---

I know that JSX is just syntactic sugar for calling `React.createElement()` and I'm not saying one or the other is better/worse, but was just wondering why you chose to use `createElement` when the React team also [recommends](https://reactjs.org/docs/react-api.html#creating-react-elements) using JSX in their docs.
> We recommend using JSX to describe what your UI should look like.

Is it preferred to use `createElement` in libraries and only use JSX in applications or is it just a personal preference?

**Comment by sebastiandedeyne** [comment]
At the moment the library doesn't have or need a build step. Introducing JSX would mean introducing a build step, adding complexity and maintenance burden to the repository. Note that we also can't really add a `dist` folder until we're going to publish on npm either.

If we would already have a build step in place, and I could easily support JSX transformation, I'd use JSX.

Alternatively, if I'd be building a template-heavy library, I'd also use JSX from th get go.

In short, adding JSX is adding a non-trivial amount of complexity to the repository, and imo it's not a worthy tradeoff just for those two lines of code.

**Comment by MichaelDeBoey** [comment]
Thanks for explaining the reasoning behind your choice @sebastiandedeyne 🙂
